pipeline-maven/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/CoveragePublisher.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    private CoverageTool buildJacocoTool(StepContext context, List<Element> events)
+            throws IOException, InterruptedException {
+        TaskListener listener = context.get(TaskListener.class);
+        FilePath workspace = context.get(FilePath.class);
+
+        List<String> patterns = new ArrayList<>();
+
+        for (Element event : events) {
+
+            Element buildElement = XmlUtils.getUniqueChildElementOrNull(event, "project", "build");
+            if (buildElement == null) {
+                if (LOGGER.isLoggable(Level.FINE))
+                    LOGGER.log(
+                            Level.FINE,
+                            "Ignore execution event with missing 'build' child:" + XmlUtils.toString(event));
+                continue;
+            }
+
+            Element pluginElt = XmlUtils.getUniqueChildElement(event, "plugin");
+            Element outputDirectoryElt = XmlUtils.getUniqueChildElementOrNull(pluginElt, "outputDirectory");
+            Element projectElt = XmlUtils.getUniqueChildElement(event, "project");
+            MavenArtifact mavenArtifact = XmlUtils.newMavenArtifact(projectElt);
+            MavenSpyLogProcessor.PluginInvocation pluginInvocation = XmlUtils.newPluginInvocation(pluginElt);
+
+            if (outputDirectoryElt == null) {
+                listener.getLogger()
+                        .println("[withMaven] No <outputDirectory> element found for <plugin> in "
+                                + XmlUtils.toString(event));
+                continue;
+            }
+            String outputDirectory = outputDirectoryElt.getTextContent().trim();
+            if (outputDirectory.contains("${project.build.directory}")) {
+                String projectBuildDirectory = XmlUtils.getProjectBuildDirectory(projectElt);
+                if (projectBuildDirectory == null || projectBuildDirectory.isEmpty()) {
+                    listener.getLogger()
+                            .println("[withMaven] '${project.build.directory}' found for <project> in "
+                                    + XmlUtils.toString(event));
+                    continue;
+                }
+                outputDirectory = outputDirectory.replace("${project.build.directory}", projectBuildDirectory);
+
+            } else if (outputDirectory.contains("${project.reporting.outputDirectory}")) {
+                String projectBuildDirectory = XmlUtils.getProjectBuildDirectory(projectElt);
+                if (projectBuildDirectory == null || projectBuildDirectory.isEmpty()) {
+                    listener.getLogger()
+                            .println("[withMaven] '${project.reporting.outputDirectory}' found for <project> in "
+                                    + XmlUtils.toString(event));
+                    continue;
+                }
+                outputDirectory = outputDirectory.replace(
+                        "${project.reporting.outputDirectory}", projectBuildDirectory + File.separator + "site");
```

Comment:
`site` should not be hardcoded, but no mention of this property in the spy log.

Will work on it when doing https://issues.jenkins.io/browse/JENKINS-54779

Reply:
Well, maybe better not to rely on some auto-maven site publication. 
better using `recordIssues`  etc..  from https://github.com/jenkinsci/warnings-ng-plugin
If people really want to use a Maven-generated HTML site, maybe it's better to let simply use htmlpublisher so nothing to hardcode here

Reply:
Hello @olamy !

I do not really get your point :-/
I do not rely on some auto-maven site publication, I rely on the `report` goal of the `jacoco-maven-plugin` which will generate an XML in some report directory, XML which will be parsed by the `coverage jenkins plugin`.

My problem here is the default value of this report directory, including the `project.reporting.outputDirectory` maven variable, defaulting to `${project.build.directory}/site`, but without an explicit value in the spy log ...
It was just a note to myself to find a way to have an explicit value for this variable, as for the `project.build.directory` one

Reply:
oops sorry for my confusion.